src/types/request-types.ts

```diff
@@ -0,0 +1,16 @@
+import { ServerScope } from 'nano'
+import { ExpressRequest } from 'serverlet/express'
+
+import { ApiKey } from './push-types'
+
+export interface DbRequest extends ExpressRequest {
+  readonly connection: ServerScope
+}
+
+export interface ApiRequest extends DbRequest {
+  readonly apiKey: ApiKey
+
+  // Taken from the Express request:
+  json: unknown
+  query: unknown
```

Comment:
Is this safe? I thought using serverlet is to eliminate loose types like this?

Reply:
The `unsafe` type *forces* you to use a cleaner, since anything else will error out. We don't actually know what these values contain (it's dynamic at runtime), so the `unsafe` type accurately captures that.